[docs] [migrate] Docs for YugabyteDB Voyager (#12820)
yugabyte/yugabyte-db: docs/content/preview/migrate/yb-voyager/migrate-steps.md

- [Prepare the source database](#prepare-the-source-database).
- [Prepare the target database](#prepare-the-target-database).

## Prepare the source database

Prepare your source database by creating a new database user, and provide it with READ access to all the resources which need to be migrated.

<ul class="nav nav-tabs nav-tabs-yb">
  <li >
    <a href="#postgresql" class="nav-link active" id="postgresql-tab" data-toggle="tab" role="tab" aria-controls="postgresql" aria-selected="true">
      <i class="icon-postgres" aria-hidden="true"></i>
      PostgreSQL
    </a>
  </li>
  <li>
    <a href="#mysql" class="nav-link" id="mysql-tab" data-toggle="tab" role="tab" aria-controls="mysql" aria-selected="false">
      <i class="icon-mysql" aria-hidden="true"></i>
      MySQL
    </a>
  </li>
  <li>
    <a href="#oracle" class="nav-link" id="oracle-tab" data-toggle="tab" role="tab" aria-controls="oracle" aria-selected="false">
      <i class="icon-oracle" aria-hidden="true"></i>
      Oracle
    </a>
  </li>
</ul>

<div class="tab-content">
  <div id="postgresql" class="tab-pane fade show active" role="tabpanel" aria-labelledby="postgresql-tab">

>     {{% includeMarkdown "./postgresql.md" %}}
>   </div>
>   <div id="mysql" class="tab-pane fade" role="tabpanel" aria-labelledby="mysql-tab">
>     {{% includeMarkdown "./mysql.md" %}}
>   </div>
>   <div id="oracle" class="tab-pane fade" role="tabpanel" aria-labelledby="oracle-tab">
>     {{% includeMarkdown "./oracle.md" %}}

Reviewer: ```suggestion
  {{% includeMarkdown "./postgresql.md" %}}
  </div>
  <div id="mysql" class="tab-pane fade" role="tabpanel" aria-labelledby="mysql-tab">
  {{% includeMarkdown "./mysql.md" %}}
  </div>
  <div id="oracle" class="tab-pane fade" role="tabpanel" aria-labelledby="oracle-tab">
  {{% includeMarkdown "./oracle.md" %}}
```

{{< /tip >}}

#### Import data status

Run the `yb-voyager import data status --export-dir ${EXPORT_DIR}` command to get an overall progress of the data import operation.

#### Import data file

If all your data files are in csv format and you have already created a schema in your target YugabyteDB, you can use the `yb-voyager import data file` command to load the data into the target table directly from the csv file(s). This command doesn’t require performing other migration steps ([export and analyze schema](#export-and-analyze-schema), [export data](#export-data), or [import schema](#import-schema) prior to import. It only requires a table present in the target database to perform the import.

<!-- ```sh
yb-voyager import data file --export-dir ${EXPORT_DIR} \
        --target-db-host ${TARGET_DB_HOST} \
        --target-db-port ${TARGET_DB_PORT} \
        --target-db-user ${TARGET_DB_USER} \
        --target-db-password ${TARGET_DB_PASSWORD:-''} \
        --target-db-name ${TARGET_DB_NAME} \
        –-data-dir “/path/to/files/dir/” \
        --file-table-map “filename1:table1,filename2:table2” \
        --delimiter “|” \
        –-has-header \
``` -->

```sh

> yb-voyager import data file--export-dir ${EXPORT_DIR} \

Author: cc @sanyamsinghal

> yb-voyager import data file--export-dir ${EXPORT_DIR} \

Reviewer 2: @ddhodge `--export-dir` is required, actually we do some bookkeeping about the state of migration to provide functionality like Resumability in case of any failure and also for faster loading by creating intermediate smaller splits of the original file before loading.

```sh
yb-voyager import data --export-dir ${EXPORT_DIR} \
        --target-db-host ${TARGET_DB_HOST} \
        --target-db-user ${TARGET_DB_USER} \
        --target-db-password ${TARGET_DB_PASSWORD} \
        --target-db-name ${TARGET_DB_NAME}
        --target-db-schema ${TARGET_DB_SCHEMA}
        --parallel-jobs 100 \
        --batch-size 250000
```

{{< tip title="Importing large datasets" >}}

When importing a very large database, run the import data command in a `screen` session, so that the import is not terminated when the terminal session stops.

If the `yb-voyager import data` command terminates before completing the data ingestion, you can re-run it with the same arguments and the command will resume the data import operation.

{{< /tip >}}

#### Import data status

Run the `yb-voyager import data status --export-dir ${EXPORT_DIR}` command to get an overall progress of the data import operation.

#### Import data file

> Use the `yb-voyager import data file` command if you have all your data files in csv format, and if you have already created a schema in your target YugabyteDB. This command allows you to load the data into the target table directly from the csv file(s). This command doesn’t require performing other phases of migration :[export and analyze schema](#export-and-analyze-schema), [export data](#export-data), or [import schema](#import-schema), prior to the import data phase. It only requires a table present in the target database to perform the import.

Reviewer 3: ```suggestion
If all your data files are in csv format and you have already created a schema in your target YugabyteDB, you can use the `yb-voyager import data file` command to load the data into the target table directly from the csv file(s). This command doesn’t require performing other migration steps ([export and analyze schema](#export-and-analyze-schema), [export data](#export-data), or [import schema](#import-schema)) prior to import. It only requires a table present in the target database to perform the import.
```
